### ISSUE TITLE
WPF VA Client: Speech recognition text is empty after the first turn

### ISSUE THREAD

**Issue by BruceHaley** [Azure-Samples/Cognitive-Services-Voice-Assistant#593]
### This issue is for a: (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```
**Background**
My Xamarin client app started failing sometime in early August. Previous versions which had worked before stopped working.

I was able to reproduce the problem using the C# WPF VA Client sample: The speech service works correctly for the first turn, then starts failing for subsequent turns.

**Steps To Reproduce**
I set up the C# EchoBot sample following the tutorial here: https://learn.microsoft.com/en-us/azure/ai-services/speech-service/tutorial-voice-enable-your-bot-speech-sdk
The code came from here: https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/02.echo-bot

I deployed EchoBot to Azure. I tested EchoBot using the Bot Framework Emulator to ensure it was working. It was.

I used the C# Windows Voice Assistant Client WPF sample application ("WVAC") found here: https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/tree/main/clients/csharp-wpf

Running WVAC and connecting to EchoBot, I followed these steps to reproduce the problem:

1. In Windows 10 open Visual Studio 2022. 
2. Open and build the WVAC solution VoiceAssistantClient.sln. 
3. Hit F5 to debug. The WVAC app window opens.
4. Click the Reconnect button. The Conversation View ("CV") shows "Hello and welcome!". A voice says the same thing.
5. Click the microphone button and say "Test 1". The CV shows "Test 1", then shows the bot's response "Echo: test 1", and a voice speaks that response. 

So far everything is working correctly.

6. Click the microphone button again and say "Test 2". 

**Actual behavior**
The CV does not show "Test 2", then after about 20 seconds it shows and speaks the bot's response "Echo:".

**Expected behavior**
The CV shows "Test 2", then shows and speaks the bot's response "Echo: test 2". 

From this point on, clicking the mic and saying something gets the same result: The utterance is not printed in the CV, and after a timeout period, the CV prints and speaks the bot's response as "Echo:", indicating that the bot received an empty string from the speech service.

Attached is a copy of the Speech SDK log output. 

How can I get the C# Windows Voice Assistant Client to work correctly with the speech service? Months ago, it used to work.

**Additional Observations**

I checked for a memory leak in Windows Voice Assistant Client using Visual Studio 2022. Results were: the objects count grew by about 1300 and the heap size grew by about 86 K each time the microphone was clicked. See the attached screen shot.

The Bot Framework Emulator works correctly with EchoBot. Differences are that the Bot Framework Emulator is written in JavaScript, not C#, and calls the speech service differently. 

Testing UWPVoiceAssistantSample shows the same problem as WPF VA Client.

[speechlog_09-26-2023_1830.txt](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/files/12733983/speechlog_09-26-2023_1830.txt)
![MemLeak-1b_2023-09-23 18_39_37-VoiceAssistantClient (Running) - Microsoft Visual Studio](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/assets/4296530/0a5832d4-b10f-4ed8-9be7-2408db5b4875)

**Comment by dargilco** [comment]
Hi @BruceHaley, this is Darren from the Speech SDK team. Thank you for your detailed comment. I'll find time today or Friday to do some experiments to confirm the issue you reported. I'll update here (likely beginning of next week) with my results.
Darren

**Comment by dargilco** [comment]
@BruceHaley I tried it this morning by building and deploying a new EchoBot and it worked fine for me. I did update the Windows Voice Assistant Client to use the latest Speech SDK version 1.32.1. Can you please do the following:
1. Clone this repo (or if you already have a clone, run 'git pull' to get the latest changes from Master branch)
2. Go to the folder \clients\csharp-wpf, open the Visual Studio solution VoiceAssistantClient.sln, and build the application
3. Lanuch VoiceAssistantClient.exe (for example, from the folder \bin\x64\Debug)
4. Make sure you see the version 1.32.1 in the top of the window (`Windows Voice Assistant Client v1.32.1`).
5. Repeat your experiment above and let me know if you still see an issue.

If you do see the same issue, please enable Speech SDK logs (enter a log file name in the `Speech SDK log file:` field in the settings of Windows Voice Assistant Client). Run the scenario you mentioned above. Close the app. Share the SDK log here.

Also, share the unique name of your Bot.

Thanks!

**Comment by BruceHaley** [comment]
@dargilco, will do.

**Comment by BruceHaley** [comment]
@dargilco, thank you for attempting a repro. As requested I cloned the latest from the repo, then built VoiceAssistantClient.sln.

My first test run reproduced the problem exactly as before. (I have been running the app using the Visual Studio debugger.)

For the second test run, I specified a log file and rebooted my machine. I clicked the mic ten times, speaking "Test 1" through "Test 10". It worked half the time. It worked correctly for turns 1, 2, 4, 6, and 9. It failed to work correctly for turns 3, 5, 7, 8, and 10. See the attached log file and screen shot.

My EchoBot name is EchoBotBH-20230822. (Messaging endpoint = https://EchoBotBH-20230822.azurewebsites.net/api/messages/) The Bot, App Service and Speech Service are all contained in the resource group EchoBotBH-20230822RG.

[speechlog_2023-10-02_1620.txt](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/files/12787216/speechlog_2023-10-02_1620.txt)
![Bug_2023-10-02 16_31_46-Windows Voice Assistant Client v1 32 1](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/assets/4296530/5d0d8ff4-ccf2-4d4d-a337-c0fa18d097ce)

**Comment by BruceHaley** [comment]
@dargilco, I ran a 3rd test run, this time executing VoiceAssistantClient.exe by double-clicking it in Windows Explorer instead of running it via the VS debugger. All 10 turns worked correctly. See attached log and screen shot. 

[speechlog_2023-10-02_1719.txt](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/files/1278
![Bug_2023-10-02 17_22_25-Windows Voice Assistant Client v1 32 1](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/assets/4296530/5a35e6f8-9fac-4007-b9bc-92e4ed60bc55)
7374/speechlog_2023-10-02_1719.txt)

I ran a 4th test run, this time in Visual Studio, "Start Without Debugging". Turns 1 through 9 succeeded. Turns 10 and 11 failed. Turns 12 through 20 succeeded. Turns 21 and 22 threw Service Timeout errors. See attached screen shot and log for that.
[Test_run_4_2023-10-02 17_42_57-Windows Voice Assistant Client v1 32 1](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/assets/4296530/ff7b8476-6b55-4d3b-8958-076440082102)
![Test_run_4_speechlog_2023-10-02_1729.txt](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/files/12787449/Test_run_4_speechlog_2023-10-02_1729.txt)

**Comment by dargilco** [comment]
Thank you @BruceHaley for doing more runs, the screenshots, and providing logs. This is very helpful. I ran the WVAC executable directly, not via VS Debugger, and I did not do so many "turns", so maybe that's why I could not reproduce your issue. Let me try again tomorrow morning. I did take a very quick look at the latest log you provided. For Turn 10, the log shows the service returning a "InitialSilenceTimeout" message which means it did not detect any speech in 5 seconds of audio that was sent to the service (it only got silent audio). I'll continue looking tomorrow and also try to run via VS Debugger.

**Comment by BruceHaley** [comment]
Hi Darren, thanks for the quick response.

For test run #5, I ran VoiceAssistantClient.exe from Windows Explorer without rebooting my machine. It errored 5 times in 15 turns. Got a Service Timeout error on turn 15.

For test run #6, I rebooted, then ran VoiceAssistantClient.exe from Windows Explorer. The first 37 turns worked without error. Every turn after that errored. "Test 38" displayed in the Conversation View, then "Hello and welcome!", then after ~60 seconds, a Service Timeout error. "Test 39" was not displayed, then after ~60 seconds, got another Service Timeout error. "Test 40" was not displayed, then "Hello and welcome!", then after ~60 seconds, got another Service Timeout error. "Test 41" was not displayed, then after ~60 seconds, got another Service Timeout error. 

For test run #7 I ran it in the VS debugger and checked memory usage. Once again the objects count increased by about 1300 and the heap size increased by about 86 K for every turn. See attached screen shot.
![Test_run_7_Memory_usage_2023-10-02 18_45_46-VoiceAssistantClient (Debugging) - Microsoft Visual Studio](https://github.com/Azure-Samples/Cognitive-Services-Voice-Assistant/assets/4296530/73cbf47a-6583-40f8-84bc-de8161057065)

VoiceAssistantClient takes longer to fail when the bare .exe is running, takes less time to fail when running without debugging in VS, and takes even less time to fail in VS when debugging. That evidence and the memory usage report from run #7 in my thinking all point to a memory leak. What do you think?

Thanks.
--Bruce

**Comment by BruceHaley** [comment]
FYI I will be out of town for most of today and will have limited ability to respond. --BH

**Comment by dargilco** [comment]
@BruceHaley The memory leak is likely from the WFP tool, and it would be good to fix it, but I don't see how that would affect your results as the leak is small. Something in thread timings changes when you run via Visual Studio, to make these errors happen more often. I tried again to get a local repo, doing 40 turns, by running .exe directly or using VS debugger, but unfortunately, I still can't get a repo. It works fine for me. I've asked for help from the services team to see if there are any errors associated with DirectLine Speech channel and your bot. There may be more than one issue here, but from the Speech SDK log it looks like at times there is no response from the service.

**Comment by BruceHaley** [comment]
@dargilco I look forward to hearing whether the services team discovers anything.

Since your tests did not repro, I am thinking maybe the problem is related to my computer environment. So I am setting up another computer to try the repro there.

Also I want to discover the source of the memory leak. If turns out to be in the SDK call to ListenOnceAsync(), then I imagine you would agree it is worth fixing no matter how small it is.

Thank you.

**Comment by BruceHaley** [comment]
@dargilco, a new development!

For test run #8, I ran WVAC in VS debugger on my same machine, but connecting to the internet from a different Seattle location, WiFi, LAN, and internet service provider.

I tested for 75 turns. All tests appeared to work correctly. I did get a "Hello and welcome!" response every so often, indicating the speech service connection had been closed and reopened. I assume that is normal speech service behavior, possibly engineered to avoid accumulating stale connections.

These results suggest the problem has to do with my ISP connection. Looks like I need to contact my ISP and ask them to troubleshoot. Dropping packets or something? IDK. Do you have any advice?

Thank you.

**Comment by dargilco** [comment]
Thank you for testing in a different environment and providing the feedback. 

Yes, if there is a memory leak in the SDK we need to fix it!

Connection to the service closes after 5 minutes of idle time (if I remember correctly... or was it 10?). The connection will automatically re-establish the next time the application calls ListenOnceAsync(), so this should be transparent to the application developer. There is a way for Bot developers to know that this is a continuing conversation with an auto-disconnected-connected client (using its unique Conversation ID), so they can suppress any greetings message that should only be sent the first time this client has connected to the Bot. The echo bot used in the voice-enable-your-bot tutorial is a very simple one that does not have this suppression, that's why you hear the greeting again after re-connection.

**Comment by BruceHaley** [comment]
@dargilco, I have isolated the root cause of the problem: 
When anything (e.g. speakers or headphones) is plugged into the j5create docking station audio IO jack, the mic input for the Windows Voice Assistant Client sample app becomes flaky. 
The workaround: Avoid using the IO audio jack on the docking station. Instead, use the audio IO jack on the computer.
Thanks for all your help.
--Bruce

**Comment by dargilco** [comment]
@BruceHaley that's interesting. Thank you for the update.
